src/vs/base/parts/ipc/common/ipc.ts

```diff
@@ -833,3 +832,29 @@ export class StaticRouter<TContext = string> implements IClientRouter<TContext>
 		return await this.route(hub);
 	}
 }
+
+export class ChannelProxy implements IServerChannel {
+
+	private service: { [key: string]: unknown };
+
+	constructor(service: unknown) {
+		this.service = service as { [key: string]: unknown };
+	}
+
+	listen<T>(_: unknown, event: string): Event<T> {
+		throw new Error(`Event not found: ${event}`);
+	}
+
+	call(_: unknown, command: string, arg?: any): Promise<any> {
+		const target = this.service[command];
+		if (typeof target === 'function') {
+			if (Array.isArray(arg)) {
```

Comment:
This will eventually break as soon as a method takes an array as its single argument.